File: validation/validation-annotation-processor/src/main/java/ru/tinkoff/kora/validation/annotation/processor/ValidTypes.java
package ru.tinkoff.kora.validation.annotation.processor;

import com.squareup.javapoet.ClassName;

public final class ValidTypes {

    private ValidTypes() {}

    public static final ClassName jsonNullable = ClassName.get("ru.tinkoff.kora.json.common", "JsonNullable");
    public static final ClassName violation = ClassName.get("ru.tinkoff.kora.validation.common", "Violation");
    public static final ClassName violationException = ClassName.get("ru.tinkoff.kora.validation.common", "ViolationException");

    public static final ClassName VALID_TYPE = ClassName.get("ru.tinkoff.kora.validation.common.annotation", "Valid");
    public static final ClassName VALIDATED_BY_TYPE = ClassName.get("ru.tinkoff.kora.validation.common.annotation", "ValidatedBy");
    public static final ClassName CONTEXT_TYPE = ClassName.get("ru.tinkoff.kora.validation.common", "ValidationContext");
    public static final ClassName VALIDATOR_TYPE = ClassName.get("ru.tinkoff.kora.validation.common", "Validator");
    public static final ClassName VIOLATION_TYPE = ClassName.get("ru.tinkoff.kora.validation.common", "Violation");
    public static final ClassName EXCEPTION_TYPE = ClassName.get("ru.tinkoff.kora.validation.common", "ViolationException");
}

File: validation/validation-symbol-processor/src/main/kotlin/ru/tinkoff/kora/validation/symbol/processor/ValidTypes.kt
package ru.tinkoff.kora.validation.symbol.processor

import com.squareup.kotlinpoet.ClassName
import com.squareup.kotlinpoet.MemberName

object ValidTypes {

    val jsonNullable = ClassName("ru.tinkoff.kora.json.common", "JsonNullable")
    val MEMBER_LIST_OF = MemberName("kotlin.collections", "mutableListOf")

    val VALID_TYPE = ClassName("ru.tinkoff.kora.validation.common.annotation", "Valid")
    val VALIDATE_TYPE = ClassName("ru.tinkoff.kora.validation.common.annotation", "Validate")
    val VALIDATED_BY_TYPE = ClassName("ru.tinkoff.kora.validation.common.annotation", "ValidatedBy")
    val CONTEXT_TYPE = ClassName("ru.tinkoff.kora.validation.common", "ValidationContext")
    val VALIDATOR_TYPE = ClassName("ru.tinkoff.kora.validation.common", "Validator")
    val VIOLATION_TYPE = ClassName("ru.tinkoff.kora.validation.common", "Violation")
    val EXCEPTION_TYPE = ClassName("ru.tinkoff.kora.validation.common", "ViolationException")
}
